packages/apollo-server-core/src/__tests__/landingPageConfigHTML.test.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  getEmbeddedExplorerHTML,
+  getEmbeddedSandboxHTML,
+} from '../plugin/landingPage/default/index';
+
+const version = '_latest';
+
+describe('Landing Page Config HTML', () => {
+  it('for embedded explorer with document, variables, headers and displayOptions provided', () => {
+    const config = {
+      includeCookies: true,
+      document: 'query Test { id }',
+      variables: {
+        option: {
+          a: 'val',
+          b: 1,
+          c: true,
+        },
+      },
+      headers: { authorization: 'true' },
+      embed: {
+        displayOptions: {
+          showHeadersAndEnvVars: true,
+          docsPanelState: 'open' as 'open',
+          theme: 'light' as 'light',
+        },
+        persistExplorerState: true,
+      },
+      graphRef: 'graph@current',
+    };
+    const expected = `
+<style>
+  iframe {
+    background-color: white;
+  }
+</style>
+<div
+style="width: 100vw; height: 100vh; position: absolute; top: 0;"
+id="embeddableExplorer"
+></div>
+<script src="https://embeddable-explorer.cdn.apollographql.com/_latest/embeddable-explorer.umd.production.min.js"></script>
+<script>
+  var endpointUrl = window.location.href;
+  var embeddedExplorerConfig = {"includeCookies":true,"document":"query Test { id }","variables":{"option":{"a":"val","b":1,"c":true}},"headers":{"authorization":"true"},"embed":{"displayOptions":{"showHeadersAndEnvVars":true,"docsPanelState":"open","theme":"light"},"persistExplorerState":true},"graphRef":"graph@current","target":"#embeddableExplorer","initialState":{"includeCookies":true,"document":"query Test { id }","variables":{"option":{"a":"val","b":1,"c":true}},"headers":{"authorization":"true"},"embed":{"displayOptions":{"showHeadersAndEnvVars":true,"docsPanelState":"open","theme":"light"},"persistExplorerState":true},"graphRef":"graph@current","displayOptions":{"showHeadersAndEnvVars":true,"docsPanelState":"open","theme":"light"}},"persistExplorerState":true};
+  new window.EmbeddedExplorer({
+    ...embeddedExplorerConfig,
+    endpointUrl,
+  });
+</script>
+`;
+    const htmlString = getEmbeddedExplorerHTML(version, config);
+    expect(htmlString).toEqual(expected);
```

Comment:
This is a good use case for jest snapshots. You can remove the expected const and just:

```ts
expect(getEmbeddedExplorerHTML(version, config)).toMatchInlineSnapshot();
```
When you run tests, it'll fill out the empty snapshot for you, which you can verify. If that snapshot ever doesn't match, Jest will tell us with a nice diff of what changed and give us an easy option to update it if appropriate.

Reply:
Bonus points: let's use an HTML snapshot serializer to give us a cleaner, unescaped snapshot. Serializers "teach" Jest how a snapshot should be printed.

```
npm i -D jest-serializer-html
```

At the top of the file we can tell Jest to use it like so:
```
expect.addSnapshotSerializer(require('jest-serializer-html'));
```